lib/semian/sysv_shared_memory.rb

```diff
@@ -0,0 +1,75 @@
+module Semian
+  module SysVSharedMemory #:nodoc:
+    @type_size = {}
+    def self.sizeof(type)
+      size = (@type_size[type.to_sym] ||= (respond_to?(:_sizeof) ? _sizeof(type.to_sym) : 0))
+      raise TypeError.new("#{type} is not a valid C type") if size <= 0
+      size
+    end
+
+    def self.included(base)
+      def base.do_with_sync(*names)
```

Comment:
@kyewei I think what he means with noop syncronize is that the SysV implementations continue to inherit from `Simple::Integer` and friends, which then look like this:

```ruby
class Semian::Simple::Integer
  # .. stuff
  def increment(delta)
    synchronize { value += delta }
  end

  private
  def synchronize
    yield
  end
  # more stuff
end
```

When this mixin is included, it just overrides it. The implementation can then be simplified when all the fallback checking is done at boot-time because you can just override it in C, and not have it in Ruby.